File: src/app/dieta.ts
import { Usuario } from "./entities/usuario";
export interface Dieta {
  id: number;
  nombre: string;
  descripcion: string;
  observaciones: string;
  objetivo: string;
  duracionDias: number|null;
  alimentos: string[];
  recomendaciones: string;
  idCliente: number|null;
  idEntrenador: number|null;
}

File: entidades/src/main/java/es/uma/informatica/practica3/entities/Dieta.java
package es.uma.informatica.practica3.entities;

import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import java.util.List;
import java.util.Set;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@EqualsAndHashCode
@ToString
@Entity
public class Dieta {
   @Id
   @GeneratedValue
   private Long id;
   private String nombre;
   private String descripcion;
   private String observaciones;
   private String objetivo;
   private Integer duracionDias;
   @ElementCollection
   private List<String> alimentos;
   private String recomendaciones;
   private Long idEntrenador;
   @ElementCollection
   private Set<Long> idClientes;

   
}
